website/static/js/filepage/index.js

```diff
@@ -225,41 +221,41 @@ var FileViewPage = {
             m('.btn-group.m-t-xs', [
                 m('.btn.btn-sm.btn-primary.file-download', {onclick: $(document).trigger.bind($(document), 'fileviewpage:download')}, 'Download')
             ]),
-            m('.btn-group.btn-group-sm.m-t-xs', [
-                m('.btn.btn-default.disabled', 'Toggle view: ')
+            m('.btn-group.btn-group-sm.m-t-xs' + (ctrl.editor ? '.btn-group' : ''), [
+               ctrl.editor ? m( '.btn.btn-default.disabled', 'Toggle view: ') : null
             ].concat(
                 m('.btn' + (ctrl.mfrIframeParent.is(':visible') ? '.btn-primary' : '.btn-default'), {
                     onclick: function (e) {
                         e.preventDefault();
-                        // atleast one button must remain enabled.
+                        // at least one button must remain enabled.
                         if (!ctrl.mfrIframeParent.is(':visible') || panelsShown > 1) {
                             ctrl.mfrIframeParent.toggle();
+                            ctrl.revisions.selected = false;
                         }
                     }
                 }, 'View')
-            ).concat(
-                panels.map(function(panel) {
-                    return m('.btn' + (panel.selected ? '.btn-primary' : '.btn-default'), {
-                        onclick: function(e) {
-                            e.preventDefault();
-                            // atleast one button must remain enabled.
-                            if (!panel.selected || panelsShown > 1) {
-                                panel.selected = !panel.selected;
-                            }
-                        }
-                    }, panel.title);
-                })
-            ))
+            ).concat([editPane()])
```

Comment:
Why use `Array.concat()` here, when we're only adding one item to the array?

I think `Array.push()` would be more explicit, plus it would avoid the creation of an Array object containing only the single element. That isn't a significant performance gain, but I generally try to avoid spurious creation of objects as a convention.